### DISCUSSION TITLE
Surface pro 9, arch linux, doesn't sleep right

### DISCUSSION THREAD

**Discussion by shafai** [linux-surface/linux-surface#1290] [General]
Hi people,
I discovered this amazing project and got rid of windows on my Surface pro 9 happily. everything is working smoothly, I have disk encryption enrolled into TPM, secure boot, etc.

The problem I have is if device goes to hibernation, power button doesn't wake it up. I have to keep pressing the power button for 10 secondish to hard power off and then pressing power button again wakes the device up. I would have lived with it if it didn't also keep draining the battery while it was in this coma state. One peculiar thing is when I put it in hibernate, the lights on the keyboard briefly turn off and on again, then turn off. which is strange.

Standby also drains the battery and if leave the laptop on my desk over night, it has lost about 50% of the battery.

I have also had exactly the same experience installing Ubuntu, so I don't think it's my lack of talent dealing with Arch.

I'm new to both Linux and surface. Want to move away from Apple, but hate to use windows. Your help is appreciated.

Cheers,
Ali

**Comment by CodeKiwi** [comment]
I have the same problem . Dammit, No responses to your question 😔

**Comment by M3GCh1na** [comment]
I just got the same. Sometimes I have to keep pressing the power button.

**Comment by Lucrtz** [comment]
Hi,

i have a Surface Pro 9 running Arch and everything works fine for me after i made some configuration.

The situation you describe sounds exactly like the [ACPI interrupt storm](https://github.com/linux-surface/linux-surface/wiki/Surface-Pro-9#acpi-interrupt-storm) problem.
You can verify this if you have **MANY** entries like `ACPI Error: No handler or method for GPE ..., disabling event (...)` in your journalctl.

The solution is:
- Add `pci=hpiosize=0` to the kernel commandline (add to GRUB_CMDLINE_LINUX_DEFAULT)
- Regenerate the GRUB configuration with `sudo grub-mkconfig -o /boot/grub/grub.cfg`
- Reboot

Another problem can also be when suspend/wake-up via the lid does not work correctly then you can blacklist the `surface_gpe` module. For this, create the file `/etc/modprobe.d/blacklist.conf` and add the line `blacklist surface_gpe` to it.

Maybe this helps you.
